### DISCUSSION TITLE
Documentation for the thin driver protocol

### DISCUSSION THREAD

**Discussion by lovetodream** [oracle/python-oracledb#124] [General]
I am planning to build a thin driver for Swift and therefor I've looked at the python thin implementation, which looks very promising. Unfortunately I do only know basic Python. I couldn't find any documentation on how to interact with oracle dbs without using OCI or JDBC online. Is there any documentation, protocol definition or something else available for the public? Or is such direct communication even intended to be done?

**Comment by cjbj** [comment]
It would be great to see a Swift thin driver. The only public information from Oracle about the network protocol is in the python-oracledb driver implementation; our team's effort is being spent on implementing features (and on a thin mode for our node-oracledb driver).  There are some 3rd party drivers that have reversed engineered the protocol. The best thing is to ask us any questions.

**Reply by lovetodream** [reply]
Very glad to hear that. I am very excited about a thin mode for the node-oracledb driver, that would be great for some of my and my employers projects. 
What's the best channel to ask you guys questions? I don't want to make unnecessary noice in this repo.

**Reply by cjbj** [reply]
Email if you like; see our profiles.

**Comment by cjbj** [comment]
@lovetodream's Swift driver is at https://github.com/lovetodream/oracle-nio and looking great.

There are also some examples at https://github.com/kicsipixel/oracle-nio-examples
